src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -53,6 +61,114 @@ struct CPU {
         if (_cores == 0) {
             _cores = _processors;
         }
+
+        /**
+         * New method to get CPU infomation and CPU map
+         */
+        _proc = sysconf(_SC_NPROCESSORS_ONLN);
+        int _cpu_mapping[_proc][CPU_MAP_USED_PROC + 1];
+        memset(_cpu_mapping, 0, _proc * (CPU_MAP_USED_PROC + 1) * sizeof(int));
+
+        auto updateProcMapping = [&](const int nproc) {
+            if (0 == _cpu_mapping[nproc][CPU_MAP_CORE]) {
+                int core_1 = 0;
+                int core_2 = 0;
+                std::string::size_type pos = 0;
+                std::string::size_type endpos = 0;
+                std::string sub_str = "";
+
+                std::ifstream cache_1_file("/sys/devices/system/cpu/cpu" + std::to_string(nproc) +
+                                           "/cache/index0/shared_cpu_list");
+                std::ifstream cache_2_file("/sys/devices/system/cpu/cpu" + std::to_string(nproc) +
+                                           "/cache/index2/shared_cpu_list");
+                std::string cache_1_info;
+                std::getline(cache_1_file, cache_1_info);
+                std::string cache_2_info;
+                std::getline(cache_2_file, cache_2_info);
+
+                if (((endpos = cache_1_info.find(',', pos)) != std::string::npos) ||
+                    ((endpos = cache_1_info.find('-', pos)) != std::string::npos)) {
+                    sub_str = cache_1_info.substr(pos, endpos);
+                    core_1 = std::stoi(sub_str);
+                    sub_str = cache_1_info.substr(endpos + 1);
+                    core_2 = std::stoi(sub_str);
+
+                    _phy_cores++;
+                    _p_cores++;
+                    _cpu_mapping[core_1][CPU_MAP_CORE] = _phy_cores;
+                    _cpu_mapping[core_2][CPU_MAP_CORE] = _phy_cores;
+                    _cpu_mapping[core_1][CPU_MAP_LOG_CORE] = _p_cores;
+                    _cpu_mapping[core_2][CPU_MAP_PHY_CORE] = _p_cores;
+
+                } else if ((endpos = cache_2_info.find('-', pos)) != std::string::npos) {
+                    sub_str = cache_2_info.substr(pos, endpos);
+                    core_1 = std::stoi(sub_str);
+                    sub_str = cache_2_info.substr(endpos + 1);
+                    core_2 = std::stoi(sub_str);
+
+                    _e_cores++;
+                    for (int m = core_1; m <= core_2; m++) {
+                        _phy_cores++;
+                        _cpu_mapping[m][CPU_MAP_CORE] = _phy_cores;
+                        _cpu_mapping[m][CPU_MAP_SMALL_CORE] = _e_cores;
```

Comment:
Why set the same value for SMALL_CORE? Could you give some clarification?

Reply:
4 Ecore will share L2 cache as group. Stream calculation need those information for better scheduling.